pkg/dynamiclifecycle/lifecycle.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dynamiclifecycle
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+type DynamicFeature string
+
+func (f DynamicFeature) String() string {
+	return string(f)
+}
+
+type DynamicFeatureLifecycle interface {
+	cell.Lifecycle
+	GetName() DynamicFeature
+	GetDependencies() []DynamicFeature
+	IsRunning() bool
+}
+
+type DefaultLifecycle struct {
```

Comment:
Do we expect multiple implementations of this? If not, I'd perhaps just call this `dynamicFeatureLifecycle`, or even just `lifecycle`.

Reply:
Also how about just embedding all this into `TableLifecycle` instead? It does mean that we'll need a "lifecycle appender" that does database updates, but it should be simpler than having the table + mutating lifecycle objects. We'd also then have transactional semantics for all the state.